livre/core/types.h

```diff
@@ -189,7 +187,7 @@ typedef std::vector< RenderBrickPtr > RenderBricks;
  * Map definitions
  */
 typedef boost::unordered_map< NodeId, dash::NodePtr > NodeIDDashNodePtrMap;
-typedef boost::unordered_map< CacheId, CacheObjectPtr > CacheMap;
+typedef boost::unordered_map< CacheId, CacheObject* > CacheMap;
```

Comment:
This seems to be the cause of the fishyness above. Why not store the shared_ptr?